[ADD] estate: created a new module

hapat-odoo opened the pull request:
Created and configured the database for the Real Estate module, defining necessary fields such as property type, tags, buyer, salesperson, and offers. Implemented list, form, and search views with groupings, and added computed fields, inverse functions, and onchange functionalities. Included 'Sold' and 'Cancel' buttons with their corresponding functionality, as well as 'Accept' and 'Refuse' buttons with related logic.

hapat-odoo committed:
[ADD]  estate:  Added a New Module estate_property

Added a new module for estate property and completed tasks 1, 2, and 3.
 I added a models folder and learned to push data into PostgreSQL using ORM.

hapat-odoo committed:
[IMP] estate:  Implemented the Security Logic

Enhanced the Security Logic for improved protection. Added the access control
rights for the estate_property.py file

hapat-odoo committed:
[IMP] estate: Added menu structure and property action

Introduced a new menu structure for the estate module, including a root menu and submenus for properties. Also added an action for managing estate properties with a corresponding view mode.

hapat-odoo committed:
[IMP] estate:  menu structure and added property views

hapat-odoo committed:
[IMP] estate:  Added state field and search view

-added  search view state field and list and form views

[IMP] estate:  Added list & form view

-added  filter for new or offer received stage

[IMP] estate:  Implemented models and views for property offers

-added  models and views for property offers field

hapat-odoo committed:
[IMP] estate: Implemented models and views for property offers, tags, and types

Implemented models and views for property offers
tags and types

hapat-odoo committed:
[IMP] estate: added automation logic

Enhanced property model and views with new fields and improved form layout

hapat-odoo committed:
[IMP] estate: Implemented offer acceptance and refusal actions

 Implemented offer acceptance and refusal actions;
added state management buttons in property views

[FIX] estate: resolve Runbot style and linting errors (Ruff/Semgrep)

This commit fixes all Runbot Check Style issues for the estate module.

[FIX] estate: Cleaned up code by removing unnecessary blank lines in model files

hapat-odoo committed:
[IMP] estate:  Implemented  constraints for price validation

-Refactored validity field to validity_days
-added constraints for price validation

hapat-odoo committed:
[IMP] estate: added Inline Views, widgets, list order

-Implemented inline views and widgets to enhance the user interface within the
Real Estate module.
-Introduced list-ordering functionality to improve the organization and sorting
of listings.
-Added attribute and option configurations to increase customization capabilities
and overall flexibility for users.

hapat-odoo committed:
[IMP] estate:  Implemented inheritance logic

-implemented inheritance in models,views
-implemented some ui changes

hapat-odoo committed:
[IMP] estate: Refine model fields and view structure

- Clean whitespace and improve formatting across model files.
- Enhance state management logic in estate_property.
- Remove unused buttons and fields to simplify property views.
- Update selection fields for improved readability and maintainability.

hapat-odoo committed:
[IMP] estate:  Implemented Interact with other modules

-Enhance property management with offer handling and invoice creation

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/tutorials/pull/1023.png)](https://mergebot.odoo.com/odoo/tutorials/pull/1023)

hapat-odoo committed:
[IMP] estate:  Implemented Kanban View

-Added Kanban view for properties and updated menu structure

hapat-odoo committed:
[IMP] awesome_owl:  Implemented new components and improve UI/JS structure

Estate:
- Updated models (property, offer, type) with improved logic and fields
- Fixed offer views and replaced deprecated attrs with modifiers for Odoo 17+
- Updated main property views for better usability
- Added cron XML for scheduled date processing
- Improved manifest metadata

Estate Accounting:
- Added account_move extension for property invoicing
- Updated estate_property accounting bridge
- Added new view estate_move.xml
- Updated access rights
- Cleaned model imports

Awesome Dashboard:
- Updated dashboard JS logic
- Added new SCSS styling
- Improved dashboard XML structure

Awesome Owl:
- Updated manifest and controllers
- Added new components: card, counter, todo_list (JS + XML)
- Updated main.js and playground code
- Added utils.js helper module

bit-odoo commented:
@hapat-odoo, multiple times i suggested the indentation issue in this PR - https://github.com/odoo/tutorials/pull/1021
But you are still making the same mistake.
https://runbot.odoo.com/runbot/batch/2246403/build/94084088

bit-odoo commented on Estate/data/date_cron.xml:
```diff
@@ -0,0 +1,10 @@
+        <field name="interval_number">1</field>
+        <field name="interval_type">days</field>
+    </record>
+</odoo>
```
Should be one empty line at the end of the file.

bit-odoo commented on Estate/security/ir.model.access.csv:
```diff
@@ -0,0 +1,6 @@
+estate.access_estate_property_offer,access_estate_property_offer,model_estate_property_offer,base.group_user,1,1,1,1
+estate.access_estate_property_type,access_estate_property_type,model_estate_property_type,base.group_user,1,1,1,1
+estate.access_estate_property_tag,access_estate_property_tag,model_estate_property_tag,base.group_user,1,1,1,1
+estate.access_res_users,access_res_users,model_res_users,base.group_user,1,1,1,1
```
Should be one empty line at the end of the file.

bit-odoo commented on Estate/views/estate_property_offer_views.xml:
```diff
@@ -0,0 +1,56 @@
+                        type="object" 
+                        icon="fa-times" 
+                        title="Refuse"
+                        invisible="status"/>
```
indentation issue.

bit-odoo commented on Estate/views/estate_property_type_views.xml:
```diff
@@ -0,0 +1,41 @@
+        <field name="name">Property Types</field>
+        <field name="res_model">estate.property.type</field>
+        <field name="view_mode">list,form</field>
+    </record>
```
it is good to have one empty line after it.

bit-odoo commented on Estate/views/estate_property_type_views.xml:
```diff
@@ -0,0 +1,41 @@
+                <field name="name"/>
+            </list>
+        </field>
+    </record>   
```
it is good to have one empty line after it.

bit-odoo commented on Estate/views/res_users_view.xml:
```diff
@@ -0,0 +1,25 @@
+                <field name="expected_price"/>
+                <field name="state"/>
+            </list>
+        </field>
```
Indantation not proper.

bit-odoo commented on Estate_account/views/estate_move.xml:
```diff
@@ -0,0 +1,12 @@
+            </xpath>
+        </field>
+    </record>
+</odoo>
```
Should be one empty line at the EOF.

bit-odoo commented on awesome_dashboard/controllers/controllers.py:
```diff
@@ -33,4 +33,3 @@ def get_statistics(self):
             },
             'total_amount': random.randint(100, 1000)
         }
-
```
unneccary diff.

bit-odoo commented on awesome_dashboard/static/src/dashboard.scss:
```diff
@@ -0,0 +1,3 @@
+.o_dashboard {
+    background-color: gray;
+}
```
Should be one empty line at the EOF.

bit-odoo commented on awesome_owl/static/src/card/card.xml:
```diff
@@ -0,0 +1,13 @@
+            </div>
+        </div>
+    </t>
+</templates>
```
Should be one empty line at the EOF.

bit-odoo commented on awesome_owl/static/src/counter/counter.xml:
```diff
@@ -0,0 +1,8 @@
+            <button class="btn btn-primary" t-on-click="increment">Increment</button>
+        </div>
+    </t>
+</templates>
```
Should be one empty line at EOF.

bit-odoo commented on awesome_owl/static/src/main.js:
```diff
@@ -9,4 +9,3 @@ const config = {
 
 // Mount the Playground component when the document.body is ready
 whenReady(() => mountComponent(Playground, document.body, config));
-
```
unneccary diff.

bit-odoo commented on awesome_owl/static/src/playground.xml:
```diff
@@ -1,10 +1,26 @@
-<?xml version="1.0" encoding="UTF-8" ?>
 <templates xml:space="preserve">
-
```
unneccary diff.

bit-odoo commented on awesome_owl/static/src/playground.xml:
```diff
@@ -1,10 +1,26 @@
     </t>
-
-</templates>
+</templates>
```
Should be one empty line at the EOF.

bit-odoo commented on awesome_owl/static/src/todo_list/todo_item.js:
```diff
@@ -0,0 +1,11 @@
+            shape: { id: Number, description: String, isCompleted: Boolean }
+        }
+    };
+}
```
Should be one empty line at the EOF.

bit-odoo commented on awesome_owl/static/src/todo_list/todo_item.xml:
```diff
@@ -0,0 +1,8 @@
+            <t t-esc="props.todo.description"/>
+        </div>
+    </t>
+</templates>
```
Should be one empty line at the EOF.

bit-odoo commented on awesome_owl/static/src/todo_list/todo_list.js:
```diff
@@ -0,0 +1,25 @@
+            ev.target.value = "";
+        }
+    }
+}
```
Should be one empty line at the EOF.

bit-odoo commented on awesome_owl/static/src/todo_list/todo_list.xml:
```diff
@@ -0,0 +1,11 @@
+            </t>
+        </div>
+    </t>
+</templates>
```
Should be one empty line at the EOF.

bit-odoo commented on awesome_owl/static/src/utils.js:
```diff
@@ -0,0 +1,8 @@
+    onMounted(() => {
+        ref.el.focus();
+    });
+}
```
Should be one empty line at the EOF.

bit-odoo reviewed (commented):
Hello,
I have made some suggestions. You are repeatedly making a basic mistake.

hapat-odoo committed:
[IMP] awesome_dashboard: Implemented JS components and fixed schema warnings

- Removed deprecated attrs usage in ListView buttons
- Ensured newline at end of XML cron file
- Improved Python validation for property states
- Corrected controller @route type to jsonrpc
